Orange/widgets/data/owpythonscript.py

```diff
@@ -655,7 +660,11 @@ def commit(self):
         self.console.new_prompt(sys.ps1)
         for out in self.outputs:
             signal = out.name
-            self.send(signal, self.console.locals.get(signal, None))
+            out_var = self.console.locals.get(signal, None)
+            if not isinstance(out_var, out.type) and out_var is not None:
+                self.Error.invalid_type(signal, out.type)
+                out_var = None
```

Comment:
The intention here is to output `None` and inform the user of that. The problem is that if there are multiple errors, the user will only be informed of the last one although all outputs are `None`.

One option is to list all errors in a single error messages, but this would become too long.

A better option is to have separate error messages. You can add them using `MessageGroup.add_message`. You can probably do the following in the class definition (that is, outside any method, in place of the existing declaration of class `Error`).

    class Error(OWWidget.Error):
        pass

    for varname, vartype in outputs:
        Error.add_message("Variable '{}' has to be of type
        invalid_type = Msg(varname, "Variable '{}' has to be of type '{}'.".format(varname, vartype.__name__)

and then raise them by something like

    getattr(self.Error, out_var)()